templates/fine-tune-function-calling/fc_utils/preprocessing.py

```diff
@@ -0,0 +1,325 @@
+"""
+Preprocessing utils for Glaive AI's function calling dataset
+"""
+
+import re
+import json
+from typing import Dict, Any, List, Optional
+import ray.data
+from fc_utils.function_extraction_utils import (
+    get_tool_calls_from_response,
+    extract_functions_from_system_msg,
+    FunctionCallNotFoundError,
+    DatasetFormat,
+)
+from fc_utils.data_format import *
```

Comment:
never do `from x import *` always be explicit about imports.

Reply:
Hmm that rule can have exceptions.  Especially with a case like here where one file has mostly constants and classes all used in the other file (preprocessing.py). I've switched to the explicit import anyways (though it's ugly and unnecessary!).